### ISSUE TITLE
McICA for SW radiation

### ISSUE THREAD

**Issue by lkluft** [CliMT/climt#67]
Hey,

this is more a question than a real issue 😉 

The LW component of RRTMG uses McICA to represent cloud variability, while the SW component is hard-coded to the non-McICA implementation.

Is there a special reason for this design decision?

Cheers,
Lukas

**Comment by lkluft** [comment]
**Update:** I just now realised that the implementation is identical to the official RRTMG source.

So it seems that one has to manually call the `mcica_subcol_sw` routine before one can use the SW-McICA.

Are there any plans to implement this as an (optional) feature for CliMT or is this out of scope?

**Comment by mcgibbon** [comment]
@JoyMonteiro can answer better whether he plans to implement this feature, but I would imagine if someone wanted to submit a PR implementing it there would be no complaints. I'd be willing to give guidance and assistance if you or someone else wants to do it.

**Comment by JoyMonteiro** [comment]
This is on my to-do list, but keeps getting pushed down the line. I will keep this issue open to remind me. We definitely want a MCICA version since RRTMG shortwave cannot handle cloud fractions otherwise (only 0 or 1 are valid inputs).

Of course, any help is always welcome :)

**Comment by lkluft** [comment]
Thanks for the quick response!

We (cc @SallyDa) will try to implement the SW-MCICA version. But we are new to Cython (and its infrastructure), so it will probably include fiddling around a bit 😉 

We will come back to you if we are completely stuck at some point.

**Comment by JoyMonteiro** [comment]
I would recommend waiting for a week before starting off on this. As you may have noticed, there is a major refactor going on, and we plan to release a version by next week. It is likely that this version will eventually become climt 1.0, and few API changes are expected.

The new API at the climt level will not change at all for single column components like RRTMG, FYI.

**Comment by mcgibbon** [comment]
I would expect it's fine for @lkluft and @SallyDa to get started once we release the paper version, which should be today or Monday at the latest. Just be aware there will be changes to merge along the way.

**Comment by mcgibbon** [comment]
Mostly the RRTM component will be insulated from the near future changes, I believe.

**Comment by JoyMonteiro** [comment]
OK, we are live!

**Comment by JoyMonteiro** [comment]
@SallyDa @lkluft, I wanted to enquire before I close this issue: are you planning to do something similar for RRTMG Longwave as well? if not, I think I will take this up in a couple of weeks.

**Comment by SallyDa** [comment]
Yes, it's almost done :)

**Comment by JoyMonteiro** [comment]
Excellent! will look forward to that then :) Closed by https://github.com/CliMT/climt/pull/83/.
